### ISSUE TITLE
iOS camera flipped

### ISSUE THREAD

**Issue by creativewax** [homuler/MediaPipeUnityPlugin#794]
### Plugin Version or Commit ID

v0.10.1

### Unity Version

2021.3.3f1

### Your Host OS

Mac OS 13.0

### Target Platform

iOS

### Target Device

IPad Pro 2021

### [iOS Only] XCode Version

15

### Build Command

Built straight out of Unity

### Bug Description

Camera is fine and orientated correctly on the default back camera, but changing it to the front camera reverses the image upside down, I wondered if this was a misconfigured setting?

### Steps to Reproduce the Bug

Tell webcam to change to front camera

### Log

No error log

**Comment by creativewax** [comment]
https://user-images.githubusercontent.com/5452341/204594085-76b8ef4d-e375-4bf0-9542-6e7f9121bfc6.mp4

Also when rotating in general also the orientation of the video stream changes

**Comment by creativewax** [comment]
I sorted the flipping issue, but not the orientation issue, the camera basically goes into an over exaggerated rotated mode when orientation changes on the device atm?

**Comment by homuler** [comment]
> the camera basically goes into an over exaggerated rotated mode when orientation changes on the device atm?

It seems to me that you implemented your sample app. Is it correct?

This repository contains a sample project and I think your problem is already solved in it.
See https://github.com/homuler/MediaPipeUnityPlugin/blob/391d7d98b127ce41ceac85ec47f6126664f1bc4e/Assets/MediaPipeUnity/Samples/Common/Scripts/Screen.cs and https://github.com/homuler/MediaPipeUnityPlugin/blob/391d7d98b127ce41ceac85ec47f6126664f1bc4e/Assets/MediaPipeUnity/Samples/Common/Scripts/GraphRunner.cs#L265-L280 for more details.

**Comment by creativewax** [comment]
Brilliant ill give that a look into, thanks ill let you know how I get on :)

**Comment by creativewax** [comment]
Finally got around to looking at this, basically this is the setup I have already. So to expand further on this, unless you start in Landscape Left mode the camera stream and tracking is out of sync, every time you rotate its like It does a double rotation, like the Mediapipe solution has flipped the camera but also the normal Unity UI has flipped everything too

**Comment by creativewax** [comment]
Sorted it with an orientation checking class, rotate the webcam and MP annotations when I have detected an orientation change and its sorted it now

**Comment by mgarbade** [comment]
@ creativewax Is this an efficient solution? Or do you now perform multiple rotations (once in your custom class and once in `GraphRunner.cs`? 
We are dealing with a similar problem and when fixing the orientation in Unity's "Edit -> ProjectSettings -> Player -> Resolution and Presentation -> Landscape Left" setting the camera image still performs a 180° rotation when flipping the tablet. Only Unity UI elements like buttons etc do not autorate anymore.

**Comment by creativewax** [comment]
@mgarbade yeah I noticed this too. In the end what I did was have it check for orientation changes in a different script and then apply offsets to both the screen component and also a custom component of mine for drawing the annontation data the way I wanted it. As weirdly depending on what orientation the device started in could have even weirder effects. There seemed to be no performance hit really doing this as I just checked it wasn’t the same orientation when trying to set so it was applied constantly
